src/App.js

```diff
@@ -22,34 +23,45 @@ export function App () {
         FILE_VIEW: 1,
     };
 
-    const urlSearchParams = new URLSearchParams(window.location.search, "?");
+    const urlHashParams = new VerbatimURLParams(window.location.hash, "#");
+    const urlSearchParams = new VerbatimURLParams(window.location.search, "?");
 
+    const [appMode, setAppMode] = useState(null);
     const [fileInfo, setFileInfo] = useState(null);
+    const [logEventIdx, setLogEventIdx] = useState(null);
+    const [prettify, setPrettify] = useState(null);
     const [theme, setTheme] = useState(THEME_STATES.DARK);
-    const [appMode, setAppMode] = useState(null);
 
     const switchTheme = (theme) => {
         localStorage.setItem("ui-theme", theme);
         document.getElementById("app").setAttribute("data-theme", theme);
         setTheme(theme);
     };
 
-    useEffect(() => {
-        console.debug("Version:", config.version);
-        const lsTheme = localStorage.getItem("ui-theme");
-        switchTheme(lsTheme === THEME_STATES.LIGHT?THEME_STATES.LIGHT:THEME_STATES.DARK);
-        if (config.defaultFileUrl) {
-            setFileInfo(config.defaultFileUrl);
+    /**
+     * Loads the file when app is rendered in this order of precedence:
+     * - File path from url if it is provided.
+     * - Default file url is used if it is provided in config file.
+     * - Provide prompt to load file.
+     */
+    const init = () => {
+        // Load the initial state of the viewer from url
+        setPrettify(urlSearchParams.get("prettify") === "true");
+        setLogEventIdx(urlHashParams.get("logEventIdx"));
+
+        const filePath = urlSearchParams.get("filePath");
+        if (filePath) {
```

Comment:
```suggestion
        if (null !== filePath) {
```

Reply:
Done.

Reply:
I dont think this is correct

Reply:
filePath is undefined if it isn't in the url

Reply:
True. Can we test for undefined then?

Reply:
Done.